### ISSUE TITLE
Solution location is incorrect if you have multiple projects in subdirectories

### ISSUE THREAD

**Issue by Teknogrebo** [premake/premake-core#119]
I am trying to migrate from Premake4 to 5, but I've hit an issue

We have a large workspace/solution that includes multiple projects in separate directories. With structure as such:

../sln/premake4.lua
../sln/src
../projA/projA_premake.lua
../projA/src/..
../projB/projB_premake.lua
../projB/src/..

sln/premake4.lua contains the Solution definition and any solution wide config options, then calls the files for all the project using dofile, like this:

solution "test"
        location "src"
        configurations "debug"
        platforms "linux"

        dofile "../projA/projA_premake.lua"
        dofile "../projB/projB_premake.lua"

With premake4 the workspace file is correctly generated in sln/src, but with premake5 it is generated in  the projB directory.  I've had a quick look at the code and I've noticed that the generate function calls path.getabsolute() which uses the current working directory.  I'm guessing that either the dofile or something else is changing the working directory, but so far haven't been able to work out where to change it to the correct directory for the location as set in the solution directive.

**Comment by starkos** [comment]
What toolset are you targeting when this happens (Visual Studio, Xcode, etc?) It might be specific to that toolset.

The [behavior of location did change between 4.x and 5.x](http://industriousone.com/topic/should-location-propagate-projects) but it sounds like you're seeing a different issue?

**Comment by Teknogrebo** [comment]
This was with codelite. I haven't tried with the other side's but I will do after the weekend.

Sent from my HTC

----- Reply message -----

**Comment by Teknogrebo** [comment]
I just tried generating a VS2015 solution from my test case and it got generated in the projB folder as well. The xcode4 project generator seems to be broken as I get an error: premake5/modules/xcode/xcode_common.lua:94: attempt to concatenate field '?' (a nil value) - but that's another issue for another time!

**Comment by Teknogrebo** [comment]
I just noticed that in each project lua file I call location("./") - obviously not knowing that that was the default.  Now all the project and solution files are produced in the location specified for the solution/workspace which isn't the way I want it, but it is at least consistent with the change you specify.  After a bit of experimentation I've discovered that if I specify any location in a project file then all solution and project files will be written to the last project specified. i.e. I can specify location("../projA") in project A config file and location("../projB") in project B's and ALL solution and project files will be generated in projB folder.
